src/client/components/table-of-contents/table-of-contents.vue

```diff
@@ -14,5 +15,78 @@
 <script>
 export default {
   props: ['items'],
+  data () {
+    return {
+      currentSection: this.items[0].slug
+    }
+  },
+  mounted () {
+    window.addEventListener('scroll', this.getClosestSection)
+  },
+  
+  beforeDestroy () {
+    window.removeEventListener('scroll', this.getClosestSection)
+  },
+
+  methods: {
+    getClosestSection () {
```

Comment:
`getClosestSection` -> `updateActiveItem`

Instead of `length` and `for`, use `this.items.forEach(item => ...`

All of the `let`s you have can actually be `const`s. As a rule of thumb, try to start out with `const` and change to `let` if you run into issues.